### ISSUE TITLE
Fix duplicate processing of the final prompt token during generation

### ISSUE THREAD

**Issue by sauravpanda** [sauravpanda/flarellm#518]
Suggested priority: P1 — generation correctness.

The default generation path processes the last prompt token twice. `Generator::generate` discards the logits returned by prefill, then `step` forwards the final prompt token again at position `prompt.len()`. The first generated token therefore uses an extra prompt token in its context.

Observed at commit `a8cb07c`. Using the existing tiny model fixture in `flare-core/tests/inference_pipeline.rs`, prompt IDs `[4, 2]`, temperature 0, and repeat penalty 1:
- Greedy selection from prefill logits returns token 3.
- `Generator::generate(..., max_tokens=1)` returns token 2.
- KV length is 3 after generation instead of the two prompt positions needed to sample the first token.

### Reproduction
Append this test to the existing inference_pipeline integration test, reusing its `make_model()` fixture:
```rust
#[test]
fn first_generated_token_matches_prefill_logits() {
    let prompt = [4u32, 2u32];
    let mut reference = make_model();
    let logits = reference.forward_prefill(&prompt);
    let expected = flare_core::sampling::sample_greedy(logits.data());

    let mut actual = make_model();
    let params = SamplingParams {
        temperature: 0.0,
        repeat_penalty: 1.0,
        ..Default::default()
    };
    let output = Generator::new(&mut actual, params)
        .generate(&prompt, 1, None, || 0.0, |_, _| true);
    assert_eq!(output[0], expected);
}
```

### Scope and acceptance criteria
- [ ] Sample the first token from prefill logits, or consistently prefill all but the last prompt token.
- [ ] Apply the invariant across core generation and browser streaming/batch entry points, including async variants.
- [ ] Cover empty/single-token prompts, BOS insertion, EOS, zero max_tokens, and greedy/sampled generation.
- [ ] Verify sequential-forward and batched-prefill generation agree within appropriate numeric tolerances.
- [ ] Make the documented default API correct without requiring callers to choose a separate healed method.

Evidence: [generate](https://github.com/sauravpanda/flarellm/blob/a8cb07cf54fa43c4e41664316498ed195fa7b018/flare-core/src/generate.rs#L421), [step](https://github.com/sauravpanda/flarellm/blob/a8cb07cf54fa43c4e41664316498ed195fa7b018/flare-core/src/generate.rs#L185), [browser default](https://github.com/sauravpanda/flarellm/blob/a8cb07cf54fa43c4e41664316498ed195fa7b018/flare-web/src/lib.rs#L1957), [existing healed method](https://github.com/sauravpanda/flarellm/blob/a8cb07cf54fa43c4e41664316498ed195fa7b018/flare-web/src/lib.rs#L1883).

Related: #234 and #236 introduced healed APIs; this issue concerns ordinary prompt processing, not partial-token completion. Workspace tests passed during review, so this needs a regression test.

**Comment by benjamin-small** [comment]
We independently encountered the same final-token duplication in browser normal streaming at `a8cb07c` while integrating SmolLM2-360M. Our public [downstream patch](https://github.com/benjamin-small/stix-meta-explorer/blob/b86a73eddfae8a5603c49e5ade8af760bc84048c/patches/flare-smollm2.patch#L13-L43) changes both `begin_stream_impl` and `begin_stream_async_impl` to prefill the first N−1 tokens and process the final token once at N−1, including the one-token case. It does not fix or claim coverage of every core/batch entry point listed in this issue.

The patch is carried in STIX Meta Explorer, not a standalone Flare fork. With our accompanying loader/GPU repairs, eight short prompts matched llama.cpp text/token IDs in CPU/async, CPU/healed, GPU/async and GPU/healed modes (32 cases, eight unique prompts). These are combined-patch smoke tests; empty prompts, zero max_tokens, and all other acceptance cases here still deserve dedicated upstream coverage. [Public validation notes](https://github.com/benjamin-small/stix-meta-explorer/blob/b86a73eddfae8a5603c49e5ade8af760bc84048c/docs/stix-agent-validation.md).

Sharing the implementation as a concrete browser-side example rather than opening a duplicate issue.
